experimental/generated/beats/fields.ecs.yml

```diff
@@ -897,6 +897,13 @@
       type: keyword
       ignore_above: 1024
       description: Unique container id.
+    - name: image.hash.sha256
+      level: extended
```

Comment:
by having a separate field for each hash type (sha256, ....) we will need to update ECS for every new hash alg in use (e.g. "blake3" becomes popular next year).   There may be an expectation that all of them are populated (ie we compute them vs just scooping the one from the Kube API).   

Pardon the last minute observation on this but this approach seems very maintenance heavy (us) vs just having   "<hashalg>:<b64 of hash>" as the value's structure for image.hash field.   This approach keeps it to one field to look in and automatically supports new hash algs, and parsing isn't that hard - most folks will be looking for an equality match, I suspect so they could type "sha256:ab1234ace..." for their search.

Reply:
I understand that perspective, and I'm fine if we decide on `image.hash`.

My thinking around `container.image.hash.sha256`:

* Follow the naming convention used by the [`hash.*` field set](https://www.elastic.co/guide/en/ecs/current/ecs-hash.html) and its reuses, like `file.hash.sha256`.
* If the `related.*` fields are being populated, [`related.hash`](https://www.elastic.co/guide/en/ecs/current/ecs-related.html#field-related-hash) helps simplify searching across the different hash fields.
* Avoid establishing `image.hash` as a leaf field

Reply:
interesting.   related.hash as an array of all the hash values doesn't seem to help as there would only be 1 entry in the array and it wouldn't be self-describing (could be sha256, sha512, ?).

I agree we should not overload "hash" to be a the existing field set or this new primitive valued one :head-explode:.

How does this sound:   Add a field to the "hash" field set, something like hash.all that holds self-describing hash values like <hash-alg>:<b64-hash-bytes>.   Sort of related.hash but co-located with hash field set and self-describing.  This would accommodate the current case of 1 hash value read from the kube API, as well as possible future use cases where we populate the different hash algs values too.  Allowed values for the hash-alg part would be the existing field names in the hash fieldset?  Thankfully they seem to align with Kubernetes use of "sha256" (same as ECS field name).

Reply:
Mike, just to clarify: you're suggesting both fields (hash.sha256 _and_ hash.all) should be set by producers, right?

Reply:
Just hash.all to start

Reply:
> > Add a field to the "hash" field set, something like hash.all that holds an array of self-describing hash values like :
> 
> I think adding `hash.all` under the `hash.*` field set would confuse users over when to populate `user.all` vs the existing fields. If capturing the hash as `<hash-alg>:<b64-hash-bytes>` is a convention used by Kubernetes, maybe `orchestrator.image.hash.all`?

`orchestrator.image.hash.all`  is exactly what I was suggesting. If we want overlay that `all` field just there (vs adding `all` to the `hash` field set in general), that works for me.  Using something other than `all` would be fine too; I'm chasing agreement on the structure concept vs naming.

Reply:
> The `related.*` fields don't help describe the values but exist to help search. Here's an example I put together to better describe their intended usage: [example](https://gist.github.com/ebeahan/8cb6cd4e679114c09556449213f6ec96).

Thanks for the example.  My concern with `related` is that used in conjunction with the associated hash.sha256 and other field values we're duplicating data. My sense is that size does matter (the smaller the better).  If we just populated `related.hash` we'd lose information (hash alg).

Reply:
Are we happy with `orchestrator.image.hash.all = ["sha256: f7988fb6c02e0ce69257d9bd9cf37ae20a60f1df7563c3a2a6abe24160306b8d"]` then?

Reply:
That works for me - no losing information (hash alg), and no duplicates in the same event.  Any concern's @ebeahan ?  Perhaps different naming?

Reply:
Works for me 👍

Reply:
I think we meant to say `container.image...` instead of `orchestrator.image...`. Just pushed that change; let me know if it's good to go.

Reply:
@ebeahan , I noticed that hash values in the TLS field set you referenced are capitalized for consistency. Should we do the same here?

Reply:
@daniel-almeida, ECS field names are always lowercase. Is that what you mean?